src/wrappers/themis/themispp/secure_comparator.hpp

```diff
@@ -33,34 +33,46 @@ namespace themispp{
       comparator_(NULL){
       res_.reserve(512);
       comparator_=secure_comparator_create();
-      if(!comparator_)
-	throw themispp::exception_t("Secure Comparator failed creating");
-      if(THEMIS_SUCCESS!=secure_comparator_append_secret(comparator_, &shared_secret[0], shared_secret.size()))
-	throw themispp::exception_t("Secure Comparator failed appending secret");	
+      if(!comparator_){
+        throw themispp::exception_t("Secure Comparator construction failed");
+      }
```

Comment:
If I remember correctly, shared secret must be not empty.

Shall we add extra check if `shared_secret` is empty and write error "Secure Comparator construction failed: must have non-empty shared secret"

Reply:
Yes, it must be (or appending will fail). JsThemis does such early checks, I'll add them here as well. Thanks for pointing it out.